extensions/security/deployment/src/main/java/io/quarkus/security/deployment/SecurityProcessor.java

```diff
@@ -148,8 +153,9 @@ void gatherSecurityChecks(BuildProducer<BeanRegistrarBuildItem> beanRegistrars,
 
             @Override
             public void register(RegistrationContext registrationContext) {
+                CompositeIndex index = CompositeIndex.create(indexBuildItem.getIndex(), beanArchiveBuildItem.getIndex());
```

Comment:
@manovotn @mkouba do we need both indexes? Or will the bean archive index already contain the app beans?

Reply:
The bean archive index always contains the index of the root app archive.

Reply:
@mkouba meaning no need for the `CompositeIndex` and using `BeanArchiveIndex` is enough, did I got it right?

Reply:
Yes, it should be enough.

Reply:
PR updated to rely on `BeanArchiveIndexBuildItem` only